### PR TITLE
[Feature] Contract::Validate(schema: Schema) - Validate with Dry Schema

### PR DESCRIPTION
**Pull request by emaglio**
Feature suggested by @swinner2 here: https://github.com/trailblazer/trailblazer/issues/187

Validation using the Dry schema is triggered when the key `schema` is not nil.

### CODE REVIEW

**Comment by emaglio** [comment]
Actually @swinner2 had another good suggestion which is to be able to validate any incoming key from options which is not necessarily nested in `params`... @marctich @apotonick what do you guys think?

File: lib/trailblazer/operation/dry_schema.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+module Trailblazer
+  class Operation
```
Comment (apotonick) [inline lib/trailblazer/operation/dry_schema.rb]: Why do we need this file? Those features are already implemented in Reform 2.3.0.rc :thinking:

Comment (apotonick) [reply lib/trailblazer/operation/dry_schema.rb]: Oh, ok, the whole point of the `Schema` is to not couple this to Reform. What we actually want to do is as follows:

1. Introduce a generic `Errors` object that is passed through operations (could be a separate gem).
2. Allow that object to deal with AMV and Dry::V results
3. Change `Contract::` macros to use that object.

I've been chatting about this with @fran-worley and I think it's time to tackle it. Also @patcoll

**Comment by emaglio** [comment]
#6 has a better solution for this (maybe)

**Pull request closed without merging**